src/views/home/components/amount-input/amount-input.view.tsx

```diff
@@ -15,25 +15,26 @@ interface AmountInputProps {
   value?: BigNumber;
   token: Token;
   balance: BigNumber;
-  fee: BigNumber;
+  fee?: BigNumber;
   onChange: (params: onChangeParams) => void;
 }
 
 const AmountInput: FC<AmountInputProps> = ({ value, token, balance, fee, onChange }) => {
   const defaultInputValue = value ? formatUnits(value, token.decimals) : "";
   const [inputValue, setInputValue] = useState(defaultInputValue);
+  const [actualFee, setActualFee] = useState<BigNumber>();
   const classes = useAmountInputStyles(inputValue.length);
-  const actualFee = token.address === ethersConstants.AddressZero ? fee : BigNumber.from(0);
 
   const updateAmountInput = useCallback(
     (amount?: BigNumber) => {
-      if (amount) {
+      if (amount && actualFee) {
         const newAmountWithFee = amount.add(actualFee);
         const isNewAmountWithFeeMoreThanFunds = newAmountWithFee.gt(balance);
         const error = isNewAmountWithFeeMoreThanFunds ? "Insufficient balance" : undefined;
-        onChange({ amount, error });
+
+        return onChange({ amount, error });
       } else {
-        onChange({});
+        return onChange({});
```

Comment:
Why add these `return` statements? `onChange` returns nothing...

Reply:
I like to be explicit on what the function returns. It's more clear to read and to understand what a function does just by taking a quick look to it.

Reply:
> I like to be explicit on what the function returns.

I like to be explicit too, the point is that this function returns nothing. Without the `return`, this was clear. Now with `return onChange()` it's no longer explicit. I need to go to the signature of `onChange` to see if I'm returning something from `updateAmountInput`.

With this change, you made the explicit implicit.

Reply:
We agreed on leaving it since the return makes explicit when the function ends.